File: sample-projects/csharp/src/De.Sots.CellarsAndCentaurs/Adapter/Persistence/CreatureEntity.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace De.Sots.CellarsAndCentaurs.Adapter.Persistence;

[Table("creatures")]
public class CreatureEntity
{
    [Key]
    public Guid Id { get; set; }

    public CreatureType Type { get; set; }

    public int HitPoints { get; set; }

    public CreatureEntity()
    {
    }

    public CreatureEntity(Guid id, CreatureType type, int hitPoints)
    {
        Id = id;
        Type = type;
        HitPoints = hitPoints;
    }
}

File: sample-projects/go/adapter/persistence/creature_entity.go
package persistence

type CreatureEntity struct {
	Id           string `db:"id"`
	CreatureType int    `db:"creature_type"`
}

func NewCreatureEntity(id string) CreatureEntity {
	return CreatureEntity{Id: id}
}

func (e CreatureEntity) Key() string {
	return e.Id
}
